### PR TITLE
Revert net461 and allow System.Collections.Immutable 5.0

### LINKED ISSUE #38: Upper bound of System.Collections.Immutable dependency needs to be raised
in order for this package to be compatible with .net 5.0 the upper bound on System.Collections.Immutable (which is currently set [here](https://github.com/YoloDev/YoloDev.Expecto.TestSdk/blob/master/Packages.props#L11)) needs to be unlocked, otherwise this package prevents usage of the 5.0 RCs (and eventually stables). Since the rc1 dropped today and has a go-live license I figured now was the time to go ahead with this issue.

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
I reverted the changes connected to the failed attempt to support net461 and was able to do created the package locally on Ubuntu.

Release notes of System.Collections.Immutable ( https://github.com/dotnet/core/blob/master/release-notes/5.0/netstandard2.1/5.0-preview5_System.Collections.Immutable.md ) suggest that only new stuff has been added to the package, so I assume that it is safe to raise the upper bond of constraint on the version of the package.

Fixes #38

### CODE REVIEW

**Comment by gdziadkiewicz** [comment]
@Alxandr can you take a look when free?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,18 +56,17 @@ jobs:
...
 
       - name: Test
-        run: dotnet msbuild build/build.proj -target:Test -p:NoRestore=true -p:VSTestNoBuild=true
+        run: dotnet msbuild build/build.proj -target:Test -p:NoRestore=true -p:VSTestNoBuild=true -v:d
 
   pack:
-    if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v')
     name: Create package
-    runs-on: windows-latest
+    runs-on: ubuntu-latest
```
Comment (Alxandr) [inline .github/workflows/ci.yml:69]: Why change this?

File: YoloDev.Expecto.TestSdk.sln [hunk 2]
```diff
@@ -1,14 +1,15 @@
+
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.30114.128
+# Visual Studio 15
+VisualStudioVersion = 15.0.26124.0
```
Comment (Alxandr) [inline YoloDev.Expecto.TestSdk.sln:4]: Why?

File: src/YoloDev.Expecto.TestSdk/main.fs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module YoloDev.Expecto.TestSdk.Main
+
+[<EntryPoint>]
+let main _ = failwithf "YoloDev.Expecto.TestSdk is not intended to be run as a program"
```
Comment (Alxandr) [inline src/YoloDev.Expecto.TestSdk/main.fs:4]: Is this file needed?

**Comment by Alxandr** [comment]
Btw; @gdziadkiewicz; would you like maintainer acces for this repo? I don't use F# anymore, so my bandwidth for this is fairly low.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gdziadkiewicz) [reply .github/workflows/ci.yml:69]: 1. When not building for net461 it is not necessary to use windows
2. ubuntu is what was used before #34 and has documented successes in completing all of the tasks

File: YoloDev.Expecto.TestSdk.sln (hunk 2 above)
Comment (gdziadkiewicz) [reply YoloDev.Expecto.TestSdk.sln:4]: No real reason - part of the revert. I can remove this part of revert.

File: src/YoloDev.Expecto.TestSdk/main.fs (hunk 3 above)
Comment (gdziadkiewicz) [reply src/YoloDev.Expecto.TestSdk/main.fs:4]: If output type of the fsproj is exe then yes.

**Comment by gdziadkiewicz** [comment]
I'm willing to help as maintainer :)

**Comment by Alxandr** [comment]
I think I just invited you correctly. Btw; looking at your build, the package creation fails, have you looked at why?

**Comment by gdziadkiewicz** [comment]
I got the invitation and accepted it. The build is failing on Create Package/Create NuGet Package with:
```
/home/runner/.dotnet/sdk/3.1.201/Sdks/NuGet.Build.Tasks.Pack/build/NuGet.Build.Tasks.Pack.targets(77,5): error MSB4018: System.ArgumentException: PackageVersion string specified 'fatal: ambiguous argument 'master': unknown revision or path not in the working tree.;Use '--' to separate paths from revisions, like this:;'git <command> [<revision>...] -- [<file>...]';0.10.2-HEAD.78+e8ee9b9' is invalid. 
```

I remember that in #34 we resolved it by disabling the step of package creation for PRs. Basing just on the error message I guess that the script that we use for generating the package versions (https://github.com/YoloDev/YoloDev.Expecto.TestSdk/blob/master/script/git-version.sh) does not generate valid versions for PRs. I will confirm if that's really the case and come back with my findings.

**Comment by gdziadkiewicz** [comment]
@Alxandr I fixed the problem by redirecting the error from git command used in `get-branch-point`  - the script is able to continue without the commit hash from it, but the error did end up in the version making it invalid. Can it go in like that?

**Comment by Alxandr** [comment]
I can probably fix the script, but just disable it for now. I'm on a phone, would you mind making an issue about the script (with link to the error) and assigning it to me so I don't forget?

**Comment by gdziadkiewicz** [comment]
I added workaround for the script problem and created an issue for permanent fix (#40).
